src/components/interface/StepsPanel.tsx

```diff
@@ -1,55 +1,133 @@
+
+  return visibleChild;
+};
+
+export function StepsPanel({
+  sequence,
+  order,
+}: {
+  sequence: string[];
+  order: OrderObject;
+}) {
   const { studyId = null } = useParams<{
     studyId: string;
   }>();
   const currentStep = useCurrentStep();
   const navigate = useNavigate();
+  const studyConfig = useStudyConfig();
+
+  const newOrder = useMemo(() => {
+    const nOrder = deepCopy(order);
+
+    // reorder based on sequence
+    nOrder.components.sort((a, b) => {
+      const keyA = typeof a === 'string' ? a : getVisibleChildComponent(sequence, a);
+      const keyB = typeof b === 'string' ? b : getVisibleChildComponent(sequence, b);
+      const idxA = keyA ? sequence.indexOf(keyA) : 10000;
+      const idxB = keyB ? sequence.indexOf(keyB) : 10000;
+      return (idxA === -1 ? 10000 : idxA) - (idxB === -1 ? 10000 : idxB);
+    });
+    return nOrder;
+  }, [order, sequence]);
 
   return (
-    <Stack spacing="xs">
-      {order.components.map((step, idx) => {
+    <div>
+      {newOrder.components.map((step, idx) => {
         if (typeof step === 'string') {
+          const task = step in studyConfig.components && studyConfig.components[step];
           return (
-            <Group key={idx}>
-              <Badge
-                key={step}
-                style={{
-                  cursor: 'pointer',
-                }}
-                color={step === currentStep ? 'red' : 'blue'}
-                onClick={() => navigate(`/${studyId}/${step}`)}
-                size="xs"
-              >
-                {step}
-              </Badge>
-            </Group>
+            <NavLink
+              active={step === currentStep}
+              key={idx}
+              label={(
+                <div
+                  style={{ opacity: sequence.indexOf(step) === -1 ? '.3' : 1 }}
+                >
+                  {step === currentStep ? <strong>{step}</strong> : step}
+                  {task && (
+                    <div
+                      style={{
+                        marginLeft: '1rem',
+                        color: '#000',
+                        display: 'none',
+                      }}
+                    >
+                      {task.description && (
+                        <Text fw={900}>
+                          Description:
+                          {' '}
+                          <Text fw={400} component="span">
+                            {task.description}
+                          </Text>
+                        </Text>
+                      )}
+                      {task.meta && <Text fw={900}>Task Meta:</Text>}
+                      {task.meta
+                        && Object.keys(task.meta).map((key) => (
+                          <Text key={key}>
+                            {key}
+                            :
+                            {' '}
+                            {(task.meta as Record<string, ReactNode>)[key]}
```

Comment:
Maybe this just needs to be cast to string, or we need to add a nice json renderer?